CMakeLists.txt

```diff
@@ -1,19 +1,36 @@
 cmake_minimum_required (VERSION 2.8.11)
 project (libpd C)
 
-option(pd_utils "Compile utilities" ON)
-option(pd_extra "Compile extras" ON)
-option(pd_multi "Compile with multiple instance support" OFF)
-option(pd_locale "Set the LC_NUMERIC number format to the default C locale" ON)
-option(pd_build_c_examples "Builds C API example programs" OFF)
+option(PD_UTILS "Compile utilities" ON)
+option(PD_EXTRA "Compile extras" ON)
+option(PD_MULTI "Compile with multiple instance support" OFF)
+option(PD_LOCALE "Set the LC_NUMERIC number format to the default C locale" ON)
+option(PD_BUILD_C_EXAMPLES "Builds C API example programs" OFF)
+
+if (WIN32)
+    option(CMAKE_THREAD_LIBS_INIT "Path to pthreads library")
+    option(PTHREADS_INCLUDE_DIR "Path to the pthreads library header files")
+    # We need pthreads.
+    # Please provide the path to the pthreads library include path and
+    # the path to the pthread library by specifying the following arguments
+    # on the CMake command-line:
+    # -DCMAKE_THREAD_LIBS_INIT:PATH=<path to library, either MSVC (for example pthreadVC2.lib) or GNUC version>
+    # -DPTHREADS_INCLUDE_DIR:PATH=<path to the pthread header files>
+    if (NOT CMAKE_THREAD_LIBS_INIT OR NOT PTHREADS_INCLUDE_DIR)
+        message(FATAL_ERROR "Please provide a path to the pthreads library and its headers.")
```

Comment:
hmm, assuming you are compiling with MSYS2 on Win32...i think msys2 will properly (provide and) find pthreads on its own, so the test seems to be over-zealous.

how about: checking whether one can use `-lpthread`?

Reply:
I did not find any way to check for -pthread compatibility. Now I only require the pthreads library location when using MSVC.
The user can still provide it if not using MSVC and will be taken into account, in case a different library is desired.